invenio_records_resources/services/records/service.py

```diff
@@ -579,3 +581,73 @@ def on_relation_update(
 
             self.reindex(identity, search_query=search_query)
         return True
+
+    @unit_of_work()
+    def create_or_update_many(self, identity, data, uow=None):
+        """Create or update a list of records.
+
+        This method takes a list of record data and creates or updates the corresponding records.
+
+        :param identity: The user identity performing the operation.
+        :param data: A list of tuples containing the record ID and record data.
+        :param uow: The unit of work to register the record operations. Defaults to None.
+
+        Returns:
+            list: A list of tuples containing the operation type ('create' or 'update'), the processed record or the record dict, and any schema errors encountered.
+        """
+        records_processed = []
+        for record_id, record_dict in data:
+            try:
+                record = self.record_cls.pid.resolve(record_id)
+
+                # Permissions
+                self.require_permission(identity, "update", record=record)
+                record_data, schema_errors = self.schema.load(
+                    record_dict,
+                    context=dict(identity=identity, pid=record.pid, record=record),
+                    raise_errors=False,
+                )
+
+                # If errors we avoid creating the record
+                if schema_errors:
+                    records_processed.append(("update", record_dict, schema_errors))
+                    continue
+
+                # Run components
+                self.run_components(
+                    "update", identity, data=record_data, record=record, uow=uow
+                )
+
+                records_processed.append(("update", record, schema_errors))
+            except PIDDoesNotExistError:
+                self.require_permission(identity, "create")
+
+                # Validate data and create record with pid
+                record_data, schema_errors = self.schema.load(
+                    record_dict, context={"identity": identity}, raise_errors=False
+                )
+
+                # If errors we avoid creating the record
+                if schema_errors:
+                    records_processed.append(("create", record_dict, schema_errors))
+                    continue
+
+                # It's the components who saves the actual data in the record.
+                record = self.record_cls.create({})
```

Comment:
would it make sense to split create_many and update_many to have dedicated methods? it would help to avoid having too much responsibilities handled in one place

Reply:
Good point! For sure there is room to improve it. I feel like it might be a bit more complicated than what we think because right now we are resolving the PID to know if a record has to be updated or created, then we should get a list with those and then pass the list of the data to 2 different funcs (update | create) + handle the exceptions there as well. I would say that we can log this as an improvement, if you agree.